### PR TITLE
Join table for visits

### PR DESCRIPTION
**Pull request by Mithi-code**
### Vet clinic database: add "join table" for visits

- Understand the different types of relationships between tables.
- Prepare complex queries that answer analytical questions.
- Use primary key & foreign key mechanisms for joining tables.
- Many-to-many relationships between tables.

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -43,3 +43,28 @@ ALTER TABLE animals
...
+    FOREIGN KEY (vet_id) REFERENCES vets (id) 
+);
+
+--Visit table
+CREATE TABLE visits (
+    animal_id INT NOT NULL,
+    vet_id INT NOT NULL,
+    date_of_visit DATE,
+    FOREIGN KEY (animal_id) REFERENCES animals (id),
+    FOREIGN KEY (vet_id) REFERENCES vets (id)
```
Comment (andersonlebon) [inline schema.sql]: image

- [ ] Kindly note that these two tables :point_up:  are violating the unique `PRIMARY KEY` in the table. you are defining two primary keys in these tables.

```sql
-- Owners table 
CREATE TABLE owners (
    id SERIAL PRIMARY KEY NOT NULL,
    full_name varchar(100),
    age INT
);
--or
CREATE TABLE owners (
    id SERIAL NOT NULL,
    full_name varchar(100),
    age INT,
   PRIMARY KEY(id)
);
 

-- Species table 
CREATE TABLE species (
    id SERIAL PRIMARY KEY NOT NULL,
    name varchar(100)
);
--or 
CREATE TABLE species (
    id SERIAL NOT NULL,
    name varchar(100),
    PRIMARY KEY(id)
);
```

This is found error :point_down: 

image

File: schema.sql [hunk 2]
```diff
@@ -43,3 +43,28 @@ ALTER TABLE animals
...
   REFERENCES owners (id);
+
+-- Creating new table vets
+CREATE TABLE vets (
+  id SERIAL PRIMARY KEY NOT NULL,
+  name VARCHAR(100) NOT NULL, 
+  age INT,
+  date_of_graduation DATE NOT NULL 
+)
+
```
Comment (andersonlebon) [inline schema.sql]: - [ ] Kindly add a `;` at the end of this query

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @Mithi-code,

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Mithi, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**